Libs/RaptureCore/src/main/java/rapture/script/reflex/ReflexRaptureScript.java

```diff
@@ -75,416 +75,380 @@
-			throw new ReflexException(e.line, message, e);
-		}
-	}
-
-	@Override
-	public String runOperation(CallingContext context, RaptureScript script, String ctx, Map<String, Object> params) {
-		try {
-			ReflexTreeWalker walker = getParserWithStandardContext(context, script.getScript(), params);
-			walker.currentScope.assign("_ctx", new ReflexValue(ctx));
-			ReflexNode res = walker.walk();
-			return res.evaluateWithoutScope(new NullDebugger()).toString();
-		} catch (RecognitionException e) {
-			String message = getErrorInfo(e.getMessage(), script, e.line, e.charPositionInLine);
-			Kernel.writeAuditEntry(EXCEPTION, 2, message);
-			throw new ReflexException(e.line, message, e);
-		}
-	}
-
-	@Override
-	public String runProgram(CallingContext context, IActivityInfo activity, RaptureScript script,
-			Map<String, Object> extraVals) {
-		return runProgram(context, activity, script, extraVals, -1);
-	}
-	
-	// how many lines to show (if possible) each side of the error
-	private static final int DEBUG_CONTEXT = 5;
-	
-	private String getErrorInfo(String message, RaptureScript script, int lineNum, int posInLine) {
-		StringBuilder msg = new StringBuilder();
-		msg.append(message).append(" in script ").append(script.getName()).append("\n");
-
-		if (lineNum > 0) {
-			String[] lines = script.getScript().split("\n");
-
-			int start = (lineNum > DEBUG_CONTEXT) ? lineNum - DEBUG_CONTEXT : 0;
-			int end = lineNum + DEBUG_CONTEXT;
-			if (end > lines.length)
-				end = lines.length;
-			while (start < end) {
-				String l = lines[start++];
-				msg.append(start).append(": ").append(l).append("\n");
-				if (start == lineNum) {
-					for (int i = -4; i < posInLine; i++)
-						msg.append("-");
-					for (int i = posInLine; i < l.length(); i++)
-						msg.append("^");
-				}
-			}
-		}
-		return msg.toString();
-	}
-
-	public String runProgram(CallingContext context, IActivityInfo activity, RaptureScript script,
-			Map<String, Object> extraVals, int timeout) {
-		String ret = null;
-		if (script == null)
-			log.info("in runProgram: RaptureScript is null");
-		else
-			try {
-				MDCService.INSTANCE.setReflexMDC(script.getName());
-				// At this point we want to collect some information about the
-				// running
-				// script and use that to generate a blob that we can
-				// record to the system blob.
-				// it will contain information about who run this script, what
-				// script it was
-				// when it was run
-				// the output from the script
-				// the instrumentation from the script
-
-				ReflexTreeWalker walker = getParserWithStandardContext(context, script.getScript(), extraVals);
-				ReflexNode res = walker.walk();
-				ScriptRunInfoCollector collector = ScriptRunInfoCollector.createServerCollector(context,
-						script.getAddressURI().getFullPath());
-				ProgressDebugger progress = (timeout < 0)
-						? new TimeoutReflexDebugger(activity, script.getScript(), timeout)
-						: new ProgressDebugger(activity, script.getScript());
```

Comment:
This was just plain wrong, and clearly never tested, If timeout < 0 TimeoutDebugger will never time out.